### PR TITLE
chore: Implement typechecking

### PR DESCRIPTION
**Pull request by doniyor2109**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,9 +13,15 @@
     "bump": "lerna version",
     "lint": "js-tools lint",
     "tdd": "jest --watch",
-    "test": "jest --coverage"
+    "test": "jest --coverage",
+    "typecheck": "lerna run typecheck"
```
Comment (umidbekkarimov) [inline package.json]: why can't we just run `tsc` from the root directory?

File: types/confusing-browser-globals.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+declare module 'confusing-browser-globals';
```
Comment (umidbekkarimov) [inline types/confusing-browser-globals.d.ts]: I think we should not add d.ts types,  just allow usage of `require` without types.

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./packages/tsconfig/tsconfig",
+  "compilerOptions": {
+    "noEmit": true,
```
Comment (umidbekkarimov) [inline tsconfig.json]: `noEmit` should be in `@superdispatch/tsconfig` already

**Comment by umidbekkarimov** [comment]
@doniyor2109 great job! but can't file structure as it were before?

**Comment by doniyor2109** [comment]
@umidbekkarimov You mean put files outside `src` folder?

**Comment by umidbekkarimov** [comment]
@doniyor2109 yes, only include types definitions

File: types/confusing-browser-globals.d.ts (hunk 2 above)
Comment (doniyor2109) [reply types/confusing-browser-globals.d.ts]: TS is reporting it.

File: package.json [hunk 4]
```diff
@@ -13,9 +13,15 @@
     "bump": "lerna version",
     "lint": "js-tools lint",
     "tdd": "jest --watch",
-    "test": "jest --coverage"
+    "test": "jest --coverage",
+    "typecheck": "yarn tsc"
```
Comment (umidbekkarimov) [inline package.json]: I think this is obsolete now, you can run `tsc` from the CI

File: packages/eslint-plugin/rules/directory-name.js [hunk 5]
```diff
@@ -26,6 +36,8 @@ module.exports = {
 
   create(context) {
     const baseDir = _.trim(
+      // TODO Remove @ts-ignore
+      // @ts-ignore
```
Comment (umidbekkarimov) [inline packages/eslint-plugin/rules/directory-name.js]: what's wrong with that one?

Comment (doniyor2109) [reply packages/eslint-plugin/rules/directory-name.js]: `eslint` seems to have missing type on `context.getCwd()`.

Maybe I will add it to type definition.

**Pull request merged**